### PR TITLE
Epic Tests

### PR DESCRIPTION
**Pull request by nameer1811**
The package is installable and the only test works. Some changes were made to the main function in LWA_bifrost.py.

### CODE REVIEW

**Comment by adampbeardsley** [comment]
@HaraKrish it would be good to get your eyes on this PR too - it will change the way LWA_Epic is installed. Hopefully it won't change how it's actually used though.

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+long_description = file: README.md
+long_description_content_type = text/markdown
+Classifiers =
+    Programming Language :: Python :: 3
+    Operating System :: Linux
+
+[options]
+packages = find:
+python_requires = >=3.6
+include_package_data = True
```
Comment (mkolopanis) [inline setup.cfg:12]: We should explicitly list our dependencies in an install_requires section. I know there is a whole install document to get people going, but being explicit about our installation requirements will help pip help users get things right.

File: .github/workflows/run_tests.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        cp $WORKDIR/LWA_EPIC/config/ASU_user.mk user.mk
+        uname -s
+        make -j 32
+        make install INSTALL_LIB_DIR="$WORKDIR/LWA_EPIC/3/envs/${ENV_NAME}/lib" INSTALL_INC_DIR="$WORKDIR/LWA_EPIC/3/envs/${ENV_NAME}/include" PYINSTALLFLAGS="--prefix=$WORKDIR/LWA_EPIC/3/envs/${ENV_NAME}"
+
+    - name: Install
+      run: |
+        pip install .
+
+    - name: Pytest Install
```
Comment (mkolopanis) [inline .github/workflows/run_tests.yml]: This name isn't really indicative of what is happening in this particular block. We're just _running_ pytest not installing it.

**Review by mkolopanis — COMMENTED** [review]
Thanks for all the hard work on this @nameer1811 and @adampbeardsley 

I have a few general comments and maybe a bit of discussion before I'm ready to merge this in.


Without an explicit setup.py this **requires** installation through `pip install .`. I'm not generally opposed to this, but it does make it impossible to have an editable install of LWA_EPIC. In general, I think it could be very useful to have editable installs for quick develop turnaround.

The only other thing that has come up with @adampbeardsley  is if we should just import LWA_EPIC _through_ the __init__.py to also allow users to  do the following.
```python
import LWA_EPIC

LWA_EPIC.main()
```
If there's a desire to have this, might as well add it now. We can always come back and tweak it.

Nice to have the test infrastructure built here now. It should be possible to keep moving to more and more modular setups and better testing as result.

File: .github/workflows/run_tests.yml (hunk 2 above)
Comment (adampbeardsley) [reply .github/workflows/run_tests.yml]:
```suggestion
    - name: Run pytest
```

File: setup.cfg (hunk 1 above)
Comment (adampbeardsley) [reply setup.cfg:12]: Hmm, I'm not super familiar with how `pip` does installations, but I think we have a bit of a conflict here. Most of our dependencies we ask the user to install using conda (see `config/epic.yml` file). So when we get to the `setup.cfg` file, do we only include the dependancies that pip installs?

Comment (mkolopanis) [reply setup.cfg:12]: I don't really see a conflict :thinking: pip/conda are just methods through which to get a dependency, but telling the thing that is actually installing your repo what is required to run your repo will ensure that you can actually run the code right? But there is a difficulty with relying on bifrost which requires an interesting installation procedure. Normally pip would just try to grab things and install them itself, but I wonder if there's a way to just tell pip "Don't try to install pacakges if we don't have them, just fail"

Comment (adampbeardsley) [reply setup.cfg:12]: yeah, that's the conflict I was talking about. I don't want pip to try to install things if we don't have them and want conda to install them. I'll look into whether there's a way to do that.

Comment (mkolopanis) [reply setup.cfg:12]: oh then I see _that_ conflict.

**Comment by adampbeardsley** [comment]
> python has an fftw package that is a wrapper for the fftw https://pypi.org/project/pyFFTW/

My commit message was a little snippy. I think think we require `pyFFTW`. I was trying to capture all our install requirements in the `setup.cfg` file, not realizing `fftw` is not actually a python package so pip can't install it.

File: .github/workflows/run_tests.yml [hunk 3]
```diff
@@ -46,7 +46,7 @@ jobs:
 
     - name: Install
       run: |
-        pip install .
+        python setup.py install
```
Comment (mkolopanis) [inline .github/workflows/run_tests.yml]: why did you change this? Pip is much better as package manager than the setuputils. I would honestly think the better way to install.

**Review by mkolopanis — COMMENTED** [review]
Looking good. One question here now.

File: .github/workflows/run_tests.yml (hunk 2 above)
Comment (mkolopanis) [reply .github/workflows/run_tests.yml]: Looks like this line got reverted back

**Comment by mkolopanis** [comment]
also do we require pyFFTW? i can't tell if we pinned that down.

File: .github/workflows/run_tests.yml (hunk 3 above)
Comment (adampbeardsley) [reply .github/workflows/run_tests.yml]: Oh, that was to make sure the `setup.py` script actually worked. I'll change it back.

Comment (adampbeardsley) [reply .github/workflows/run_tests.yml]:
```suggestion
        pip install .
```

**Comment by adampbeardsley** [comment]
We do not require pyFFTW.

**Review by mkolopanis — APPROVED** [review]
This is a good start. We can at least check that EPIC will run with proposed changes. Maybe we can make some todos for actual testing.
